### PR TITLE
Adding Birch

### PR DESCRIPTION
**Pull request by Silun**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Silun** [comment]
@conda-forge/staged-recipes I think this is ready for review!

File: recipes/birch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (BastianZim) [inline recipes/birch/meta.yaml:20]: Are you sure that a compiler is required? This seems like pure python.

File: recipes/birch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - strct
```
Comment (BastianZim) [inline recipes/birch/meta.yaml:26]:
```suggestion
    - strct >=0.0.26
```

File: recipes/birch/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (BastianZim) [inline recipes/birch/LICENSE.txt:1]: This can be removed, the license is already included in the tar ball.

File: recipes/birch/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/shaypal5/birch
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.txt
```
Comment (BastianZim) [inline recipes/birch/meta.yaml:40]:
```suggestion
  license_file: LICENSE
```

**Review by BastianZim — CHANGES_REQUESTED** [review]

File: recipes/birch/meta.yaml (hunk 1 above)
Comment (Silun) [reply recipes/birch/meta.yaml]: You are correct, that shouldn't have been there.

**Comment by Silun** [comment]
@BastianZim I implemented your requested changes, thank you!

**Pull request merged**